mongo/filterstore.go

```diff
@@ -2,66 +2,90 @@ package mongo
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"math"
 	"time"
 
 	"github.com/ONSdigital/dp-filter-api/config"
 	"github.com/ONSdigital/dp-filter-api/filters"
 	"github.com/ONSdigital/dp-filter-api/models"
 	"github.com/ONSdigital/dp-healthcheck/healthcheck"
-	mongolib "github.com/ONSdigital/dp-mongodb"
-	mongohealth "github.com/ONSdigital/dp-mongodb/health"
-	"github.com/globalsign/mgo"
-	"github.com/globalsign/mgo/bson"
+	dpMongoHealth "github.com/ONSdigital/dp-mongodb/v2/pkg/health"
+	dpMongoDriver "github.com/ONSdigital/dp-mongodb/v2/pkg/mongodb"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+const (
+	connectTimeoutInSeconds = 5
+	queryTimeoutInSeconds   = 15
 )
 
 // FilterStore containing all filter jobs stored in mongodb
 type FilterStore struct {
-	Session           *mgo.Session
-	host              string
-	db                string
-	filtersCollection string
-	outputsCollection string
-	healthCheckClient *mongohealth.CheckMongoClient
+	URI               string
+	Database          string
+	FiltersCollection string
+	OutputsCollection string
+	Connection        *dpMongoDriver.MongoConnection
+	Username          string
+	Password          string
+	CAFilePath        string
+	healthCheckClient *dpMongoHealth.CheckMongoClient
+}
+
+func NewMongoTimestamp(t time.Time, c uint32) (int64, error) {
+	u := t.Unix()
+	if u < 0 || u > math.MaxUint32 {
+		return -1, errors.New("invalid value for time")
+	}
+
+	i := int64(u<<32 | int64(c))
+
+	return int64(i), nil
 }
 
 // CreateFilterStore which can store, update and fetch filter jobs
 func CreateFilterStore(cfg config.MongoConfig, host string) (*FilterStore, error) {
-	session, err := mgo.Dial(cfg.BindAddr)
+	mongoConnection, err := dpMongoDriver.Open(&dpMongoDriver.MongoConnectionConfig{
+		ConnectTimeoutInSeconds: connectTimeoutInSeconds,
+		QueryTimeoutInSeconds:   queryTimeoutInSeconds,
+
+		Username:        cfg.Username,
+		Password:        cfg.Password,
+		ClusterEndpoint: cfg.BindAddr,
+		Database:        cfg.Database,
+		IsSSL:           cfg.IsSSL,
+	})
```

Comment:
can we also add these settings:
```
		IsWriteConcernMajorityEnabled: shouldEnableWriteConcern,
		IsStrongReadConcernEnabled:    shouldEnableReadConcern,
```
this is for enabling strong read and write concerns.
ref: 
https://github.com/ONSdigital/dp-mongodb/blob/main/v2/pkg/mongodb/driver.go#L81-L91